.github/workflows/compile.yml

```diff
@@ -18,7 +18,28 @@ jobs:
         run: |
           sudo apt-get update -q
           sudo apt-get remove -y clang
-          sudo apt-get install -y clang-10 autoconf-archive flex bison libjson-c-dev libxen-dev
+          sudo apt-get install -y clang-10 autoconf-archive flex bison libjson-c-dev wget git bcc bin86 gawk bridge-utils iproute2 libcurl4-openssl-dev bzip2 libpci-dev build-essential make gcc clang libc6-dev libc6-dev-i386 linux-libc-dev zlib1g-dev libncurses5-dev patch libvncserver-dev libssl-dev libsdl-dev iasl libbz2-dev e2fslibs-dev git-core uuid-dev ocaml libx11-dev bison flex ocaml-findlib xz-utils gettext libyajl-dev libpixman-1-dev libaio-dev libfdt-dev cabextract libglib2.0-dev autoconf automake libtool libjson-c-dev libfuse-dev liblzma-dev autoconf-archive kpartx python3-dev python3-pip golang python-dev libsystemd-dev
+      - name: Get Xen version
+        id: get-xen-hash
+        run: |
+          echo ::set-output name=XEN_HASH::$(cd xen && git describe --always)
+      - name: Cache Xen debball
+        id: cache-xen
+        uses: actions/cache@v2
+        with:
+           path: 'xen/dist/*'
+           key: ${{ steps.get-xen-hash.outputs.XEN_HASH }}
+      - name: Create Xen debball
+        if: steps.cache-xen.outputs.cache-hit != 'true'
+        run: |
+          git submodule update --init xen
+          cd xen
+          ./configure --enable-githttp --disable-pvshim
+          make -j4 debball
+          cd ..
+      - name: Install Xen debball
+        run: |
+          sudo dpkg -i xen/dist/xen-upstream-4.15.0-rc.deb
```

Comment:
Probably better not to hardcode the Xen version and just use `dpkg -i xen/dist/*.deb`

Reply:
yup, just omitted that